container/attribute.go

```diff
@@ -4,75 +4,99 @@ import (
 	"github.com/nspcc-dev/neofs-api-go/v2/container"
 )
 
-type (
-	Attribute  container.Attribute
-	Attributes []Attribute
-)
-
-// NewAttribute creates and initializes blank Attribute.
+// Attribute represents v2-compatible container attribute.
 //
-// Defaults:
-//  - key: "";
-//  - value: "".
-func NewAttribute() *Attribute {
-	return NewAttributeFromV2(new(container.Attribute))
-}
+// Attribute is mutually compatible with github.com/nspcc-dev/neofs-api-go/v2/container.Attribute
+// message. See ReadFromV2 / WriteToV2 methods.
+//
+// Instances can be created using built-in var declaration.
+//
+// Note that direct typecast is not safe and may result in loss of compatibility:
+// 	_ = Attribute(container.Attribute{}) // not recommended
+type Attribute container.Attribute
 
-func (a *Attribute) SetKey(v string) {
-	(*container.Attribute)(a).SetKey(v)
-}
+// Attributes groups container attributes.
+type Attributes []Attribute
 
-func (a *Attribute) SetValue(v string) {
-	(*container.Attribute)(a).SetValue(v)
+// ReadFromV2 reads Attribute from the container.Attribute message.
+//
+// See also WriteToV2.
+func (a *Attribute) ReadFromV2(m container.Attribute) {
+	*a = Attribute(m)
 }
 
-func (a *Attribute) Key() string {
-	return (*container.Attribute)(a).GetKey()
+// WriteToV2 writes Attribute to the container.Attribute message.
+// The message must not be nil.
+//
+// See also ReadFromV2.
+func (a Attribute) WriteToV2(m *container.Attribute) {
+	*m = (container.Attribute)(a)
 }
 
-func (a *Attribute) Value() string {
-	return (*container.Attribute)(a).GetValue()
+// Key returns attribute's key.
+//
+// Zero Attribute has empty key.
+//
+// See also SetKey.
+func (a Attribute) Key() string {
+	v2 := (container.Attribute)(a)
+	return v2.GetKey()
 }
 
-// NewAttributeFromV2 wraps protocol dependent version of
-// Attribute message.
+// SetKey sets attribute's key.
 //
-// Nil container.Attribute converts to nil.
-func NewAttributeFromV2(v *container.Attribute) *Attribute {
-	return (*Attribute)(v)
+// See also Key.
+func (a *Attribute) SetKey(v string) {
+	(*container.Attribute)(a).SetKey(v)
 }
 
-// ToV2 converts Attribute to v2 Attribute message.
+// Value returns attribute's value.
 //
-// Nil Attribute converts to nil.
-func (a *Attribute) ToV2() *container.Attribute {
-	return (*container.Attribute)(a)
+// Zero Attribute has empty value.
+//
+// See also SetValue.
+func (a Attribute) Value() string {
+	v2 := (container.Attribute)(a)
+	return v2.GetValue()
 }
 
-func NewAttributesFromV2(v []container.Attribute) Attributes {
-	if v == nil {
-		return nil
-	}
+// SetValue sets attribute's value.
+//
+// See also Value.
+func (a *Attribute) SetValue(v string) {
+	(*container.Attribute)(a).SetValue(v)
+}
 
-	attrs := make(Attributes, len(v))
-	for i := range v {
-		attrs[i] = *NewAttributeFromV2(&v[i])
+// ReadFromV2 reads Attributes from the []container.Attribute.
+// Attributes slice must not be nil.
```

Comment:
Did you mean `Attributes`? Otherwise if u meant parameter then statement differs with the implementation.

Reply:
hmmm, it was copied from other methods, but yes it is the senseless comment for that method, removed